ts/features/bonus/cgn/screens/merchants/CgnMerchantDetailScreen.tsx

```diff
@@ -67,7 +61,12 @@ const CgnMerchantDetailScreen: React.FunctionComponent<Props> = (
   props: Props
 ) => {
   const { merchantDetail, requestMerchantDetail } = props;
-  const merchantID = props.navigation.getParam("merchantID");
+  const route =
+    useRoute<
```

Comment:
Instead of using `useRoute` here, what do you think about adding to the type `Props` the `IOStackNavigationRouteProps<CgnDetailsParamsList, "CGN_MERCHANTS_DETAIL">` and getting `route` from the props like `const { merchantDetail, requestMerchantDetail, route } = props;`? 🤔 In this way the `CgnDetailsParamsList` remain the single source of truth for the parameters.

Reply:
I get your point, I think this can mainly relate to the fight about hooks vs prop injection I don't have a strong position, anyway the `useRoute` is typed at the same way of the related route inside `CgnDetailsParamsList` using `CgnMerchantDetailScreenNavigationParams`, what do you think?

Reply:
@CrisTofani My 0.2c for this is: if you are on a "screen" component, you are not actually doing "props drilling" because `react-navigation` already adds the props by default to the screen. In this case you already have the `route` parameter inside the prop, you're just adjusting the type correctly (if you access `props.route` without adding the type you won't get a runtime error, because the `route` param is actually there). The `useRoute` hook in my mind should be used on a component that must be added to the screen, in order to avoid the prop drilling from the screen to that component.

Btw this is just a design discussion, both the solutions are 100% fine, so I don't have a strong opinion either 😂 If you still think that `useRoute` it's a better approach I'm totally fine with that and feel free to mark this comment as resolved 🙂